src/components/Image.js

```diff
@@ -2,14 +2,64 @@
 import * as React from "react";
 import ImageZoom from "react-medium-image-zoom";
 import styled from "styled-components";
+import extract from "png-chunks-extract";
 import type { SlateNodeProps as Props } from "../types";
 
 type State = {
   hasError?: boolean,
+  imageWidth?: number,
 };
 class Image extends React.Component<Props, State> {
   state = {
     hasError: false,
+    imageWidth: undefined,
+  };
+
+  /**
+   * Detect if PNG image is in retina resolution.
+   *
+   * PNG Chunk documentation: https://www.w3.org/TR/PNG-Chunks.html
+   */
+  onImgLoad = async (ev: SyntheticEvent<HTMLImageElement>) => {
```

Comment:
I guess you have no desire to pull this into it's own module? I'd probably move into it's own lib file at least.

Will test thoroughly this w/e.

Reply:
Moved to a lib file. Not particularly fond of publishing 10 lines as a module :)